File: src/Plotly.NET/Projection.fs
namespace Plotly.NET

open DynamicObj
open System


type ProjectionDimension () =
    inherit DynamicObj () 

    static member init 
        (
            ?Opacity    : float,
            ?Scale      : float,
            ?Show       : bool
        ) =
            ProjectionDimension()
            |> ProjectionDimension.style
                (
                    ?Opacity    = Opacity,
                    ?Scale      = Scale  ,
                    ?Show       = Show   
                )

    static member style
        (
            ?Opacity    : float,
            ?Scale      : float,
            ?Show       : bool
        ) =

            fun (projectionDimension: ProjectionDimension) ->
                
                Opacity |> DynObj.setValueOpt projectionDimension "opacity"
                Scale   |> DynObj.setValueOpt projectionDimension "scale"
                Show    |> DynObj.setValueOpt projectionDimension "show"

                projectionDimension

type Projection () =
    inherit DynamicObj () 

    static member init 
        (
            ?X: ProjectionDimension,
            ?Y: ProjectionDimension,
            ?Z: ProjectionDimension
        ) =
            Projection()
            |> Projection.style
                (
                    ?X = X,
                    ?Y = Y,
                    ?Z = Z
                )

    static member style
        (
            ?X: ProjectionDimension,
            ?Y: ProjectionDimension,
            ?Z: ProjectionDimension
        ) =

            fun (projection: Projection) ->
                
                X   |> DynObj.setValueOpt projection "x"
                Y   |> DynObj.setValueOpt projection "y"
                Z   |> DynObj.setValueOpt projection "z"

                projection

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/Projection.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.ProjectionDimension InitProjectionDimension
        (
            Optional<double> Opacity = default,
            Optional<double> Scale = default,
            Optional<bool> Show = default
        ) 
            =>
                Plotly.NET.TraceObjects.ProjectionDimension.init(
                    Opacity: Opacity.ToOption(),
                    Scale: Scale.ToOption(),
                    Show: Show.ToOption()
                );
    public static Plotly.NET.TraceObjects.Projection InitProjection
        (
            Optional<Plotly.NET.TraceObjects.ProjectionDimension> X = default,
            Optional<Plotly.NET.TraceObjects.ProjectionDimension> Y = default,
            Optional<Plotly.NET.TraceObjects.ProjectionDimension> Z = default
        ) 
            =>
                Plotly.NET.TraceObjects.Projection.init(
                    X: X.ToOption(),
                    Y: Y.ToOption(),
                    Z: Z.ToOption()
                );
}

